src/Util.cc

```diff
@@ -711,12 +711,15 @@ std::optional<math::Vector3d> sphericalCoordinates(Entity _entity,
 
   // lat / lon / elevation in rad / rad / m
   auto rad = sphericalCoordinatesComp->Data().PositionTransform(
-      xyzPose.Pos(),
-      math::SphericalCoordinates::LOCAL2,
+      math::CoordinateVector3::Metric(xyzPose.Pos()),
+      math::SphericalCoordinates::LOCAL,
       math::SphericalCoordinates::SPHERICAL);
 
+  if (!rad.has_value())
```

Comment:
```suggestion
  if (!rad.has_value() || !rad->IsSpherical())
```
because `rad->Lat()` would throw if the coord is not spherical... this might seem overcautious, but the API was designed with this kind of double-checks of the results in mind...

Reply:
[9f1c0db](https://github.com/gazebosim/gz-sim/pull/2535/commits/9f1c0dbbcc50e89ae9a605811e9cbb18cc460f90)